code/Model/SiteTree.php

```diff
@@ -2712,24 +2719,30 @@ protected function getClassDropdown()
      */
     public function allowedChildren()
     {
-        $allowedChildren = array();
-        $candidates = $this->stat('allowed_children');
-        if ($candidates && $candidates != "none" && $candidates != "SiteTree_root") {
-            foreach ($candidates as $candidate) {
-                // If a classname is prefixed by "*", such as "*Page", then only that class is allowed - no subclasses.
-                // Otherwise, the class and all its subclasses are allowed.
-                if (substr($candidate, 0, 1) == '*') {
-                    $allowedChildren[] = substr($candidate, 1);
-                } elseif ($subclasses = ClassInfo::subclassesFor($candidate)) {
-                    foreach ($subclasses as $subclass) {
-                        if ($subclass == 'SiteTree_root' || singleton($subclass) instanceof HiddenClass) {
-                            continue;
+        if (isset(static::$_allowedChildren[get_class($this)])) {
+            $allowedChildren = static::$_allowedChildren[get_class($this)];
+        } else {
+            $allowedChildren = array();
+            $candidates = $this->stat('allowed_children');
+            if ($candidates && $candidates != "none" && $candidates != "SiteTree_root") {
+                foreach ($candidates as $candidate) {
+                    // If a classname is prefixed by "*", such as "*Page", then only that class is allowed - no subclasses.
+                    // Otherwise, the class and all its subclasses are allowed.
+                    if (substr($candidate, 0, 1) == '*') {
+                        $allowedChildren[] = substr($candidate, 1);
+                    } elseif ($subclasses = ClassInfo::subclassesFor($candidate)) {
+                        foreach ($subclasses as $subclass) {
+                            if ($subclass == 'SiteTree_root' || singleton($subclass) instanceof HiddenClass) {
+                                continue;
+                            }
+                            $allowedChildren[] = $subclass;
                         }
-                        $allowedChildren[] = $subclass;
                     }
                 }
             }
+            static::$_allowedChildren[get_class($this)] = $allowedChildren;
         }
+        $this->extend('updateAllowedChildren', $allowedChildren);
```

Comment:
@dhensby If a user wants to have some dynamic functionality in there based on an instance value.
The basic response from `SiteTree:getAllowedChildren()` won't change from one instance to another, only by class but the extended could change